src/sync/sync_config.hpp

```diff
@@ -43,7 +45,11 @@ using SyncBindSessionHandler = void(const std::string&,          // path on disk
                                     std::shared_ptr<SyncSession> // the session which should be bound.
                                     );
 
-using SyncSessionErrorHandler = void(std::shared_ptr<SyncSession>, int error_code, std::string message, SyncSessionError);
+using SyncSessionErrorHandler = void(std::shared_ptr<SyncSession>,
+                                     std::error_code error_code,
+                                     bool is_fatal,
```

Comment:
This method signature is starting to get slightly confusing, especially because there seem to be some dependencies between the arguments not really expressed here, e.g. isn't the `SyncSystemError` determined by the `error_code`? Also only a subset of `error_code`s are fatal? Would it make sense to wrap of all of these except `shared_ptr` in an `ObjectServerError` class instead?